docs/tutorials/add-prisma-database.md

```diff
@@ -0,0 +1,359 @@
+  const updatedRecord = await prisma.urlLike.upsert({
+    where: { pathname },
+    update: { count: { increment: 1 } },
+    create: { pathname, count: 1 },
+  });
+
+  return new Response(JSON.stringify(updatedRecord), {
+    headers: { 'Content-Type': 'application/json' },
+  });
+}
+```
+
+Let's declare the route in Seven.
+First, create a {file}`config` folder in your add-on at {file}`package/<addon-name>/config`.
+You'll store all your add-on configurations in this folder.
+
+Then, create a file in this folder {file}`package/<add-on-name>/config/routes.ts` with the following code.
+
+```{code-block} ts
+:caption: packages/\<add-on-name>/config/routes.ts
+import type { ConfigType } from '@plone/registry';
+
+export default function install(config: ConfigType) {
+  config.registerRoute({
+    type: 'route',
+    path: '@likes/*',
+    file: '<add-on-name>/routes/api.likes.ts',
+    options: {
+      id: 'likes',
+    },
+  });
+
+  return config;
+}
+```
+
+Replace `<add-on-name>` with the actual package directory name you generated with cookieplone before using the snippets below.
+
+Wire it up in your add-on's {file}`package/<add-on-name>/index.ts` file:
+
+```{code-block} ts
+:caption: packages/\<add-on-name>/index.ts
+import type { ConfigType } from '@plone/registry';
+import installRoutes from './config/routes';
+
+export default function install(config: ConfigType) {
+  installRoutes(config);
+
+  return config;
+}
+```
+
+Accessing the route with a GET request, for example, `/@likes/some/url/path`, will return the number of likes for that current URL.
+Sending a POST request will increment the like count for that URL.
+
+## Create a likes slot
+
+Create a new file named {file}`LikeButton.tsx` in the {file}`slots` folder of your add-on ({file}`package/<add-on-name>/slots/LikeButton.tsx`).
+
+```{code-block} tsx
+:caption: packages/\<add-on-name>/slots/LikeButton.tsx
+import type { RootLoader } from 'seven/app/root';
+import { Button } from '@plone/components/quanta';
+import { useFetcher, useParams, useRouteLoaderData } from 'react-router';
+
+export default function LikeButton() {
+  const fetcher = useFetcher();
+  const params = useParams();
+  const rootData = useRouteLoaderData<RootLoader>('root');
+  const pathname = params['*'] ? `/${params['*']}` : '/';
+  const rootCount = rootData?.likes?.count ?? 0;
+  const fetcherRecord = fetcher.data as { count?: number } | undefined;
+  const fetcherCount =
+    typeof fetcherRecord?.count === 'number' ? fetcherRecord.count : undefined;
+  const likeCount = fetcherCount ?? rootCount;
+
+  return (
+    <div className="text-center mt-10">
+      <fetcher.Form
+        method="post"
+        action={`/@likes${pathname}`}
+        className="inline"
+      >
+        <Button
+          variant="primary"
+          accent
+          type="submit"
+          isDisabled={fetcher.state === 'submitting'}
+        >
+          👍 Like ({likeCount})
```

Comment:
Sphinx barfs on emojis. See https://app.readthedocs.org/projects/volto/builds/29831361/#288836463--291.

Maybe use the HTML entity `&#x1F44D;` instead?

Reply:
Done. It does not render in sphinx, but it does in the app.